bundles/org.openhab.binding.salus/src/main/java/org/openhab/binding/salus/internal/rest/SalusApi.java

```diff
@@ -0,0 +1,224 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.salus.internal.rest;
+
+import static java.util.Objects.requireNonNull;
+import static org.openhab.binding.salus.internal.rest.ApiResponse.error;
+
+import java.time.Clock;
+import java.time.LocalDateTime;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The SalusApi class is responsible for interacting with a REST API to perform various operations related to the Salus
+ * system. It handles authentication, token management, and provides methods to retrieve and manipulate device
+ * information and properties.
+ *
+ * @author Martin Grześlowski - Initial contribution
+ */
+@NonNullByDefault
+public class SalusApi {
+    private static final int MAX_TIMES = 3;
+    private final Logger logger;
+    private final String username;
+    private final char[] password;
+    private final String baseUrl;
+    private final RestClient restClient;
+    private final GsonMapper mapper;
+    @Nullable
+    private AuthToken authToken;
+    @Nullable
+    private LocalDateTime authTokenExpireTime;
+    private final Clock clock;
+
+    public SalusApi(String username, char[] password, String baseUrl, RestClient restClient, GsonMapper mapper,
+            Clock clock) {
+        this.username = requireNonNull(username, "username");
```

Comment:
All these `requireNonNull` should not be needed as the class is allready annotated as `NonNullByDefault`.